packages/flutter_tools/lib/src/android/android_builder.dart

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import '../base/common.dart';
+import '../base/context.dart';
+import '../build_info.dart';
+import '../globals.dart';
+import '../project.dart';
+
+import 'android_sdk.dart';
+import 'gradle.dart';
+
+/// The builder in the current context.
+AndroidBuilder get androidBuilder => context.get<AndroidBuilder>() ?? AndroidBuilderImpl();
+
+/// Provides the methods to build Android artifacts.
+abstract class AndroidBuilder {
+  /// Builds an AAR artifact.
+  Future<void> buildAar({
+    @required FlutterProject project,
+    @required AndroidBuildInfo androidBuildInfo,
+    @required String target,
+    @required String outputDir,
+  });
+
+  /// Builds an APK artifact.
+  Future<void> buildApk({
+    @required FlutterProject project,
+    @required AndroidBuildInfo androidBuildInfo,
+    @required String target,
+  });
+
+  /// Builds an App Bundle artifact.
+  Future<void> buildAab({
+    @required FlutterProject project,
+    @required AndroidBuildInfo androidBuildInfo,
+    @required String target,
+  });
+}
+
+/// Default implementation of [AarBuilder].
+class AndroidBuilderImpl extends AndroidBuilder {
+  AndroidBuilderImpl();
+
+  /// Builds the AAR and POM files for the current Flutter module or plugin.
+  @override
+  Future<void> buildAar({
+    @required FlutterProject project,
+    @required AndroidBuildInfo androidBuildInfo,
+    @required String target,
+    @required String outputDir,
+  }) async {
+    if (!project.android.isUsingGradle) {
+      throwToolExit(
+          'The build process for Android has changed, and the current project configuration\n'
```

Comment:
This message will be printed with `printError()`, which should take care of putting line breaks an a good place taking the user's terminal width into account. Whitespace around the URL below makes sense, but otherwise we can let `printError()` decide where to put newlines.

Reply:
oh awesome! Done